Fix the TELEMETRY message + add connection hint (#42)
neo4j/docs-bolt: modules/ROOT/pages/bolt/message.adoc

=== Request message `TELEMETRY`

The `TELEMETRY` message contains an integer representing which driver API was used.

> This message is new in version *5.4*.

Reviewer: We can also add `label:new[Introduced in 5.4]` beside heading to have a label show up. It looks much nicer, but it would also be nice to have them throughout the whole docset. We can have a conversation on this.

![Screenshot from 2023-10-06 08-29-38](https://github.com/neo4j/docs-bolt/assets/114478074/9a5599fe-e89a-454b-8999-3c9ab7ca359c)

Author: I really like this tag. But I'd go for consistency. Can these labels only be applied to headlines? Because there are also fields and details that changed withing messages.

Reviewer: They can be applied to any element (paragraphs, table entries, etc), they get attached to its nearest asciidoc element (i.e. you can write
```
== Section header
label:new[Introduced in 5.4]
```
and the label still belongs to the header).

Reviewer: I don't have a strong opinion over whether this should happen here or later in the whole manual. I just wanted to point it out :)

Author: I created an issue (#44) to take care of that in a separate PR.

The telemetry information is stored on the server's metrics system.
The client receives a `SUCCESS` response, unless it sends an invalid value for the `api` field, which results in a `FAILURE` response.

This message is stored into the metrics system only if telemetry is enabled on both the server and the driver.
If only enabled on the driver, the message will still be sent and a `SUCCESS` will be returned, but nothing will be stored into the server's metrics system.

> === Request message `TELEMETRY`

Reviewer: :warning: 

This section got duplicated!

- `2` -- Implicit transaction
- `3` -- Driver-level `execute_query()`

*Detail messages:*

No detail messages should be returned.

*Valid summary messages:*

** `SUCCESS`
** `FAILURE`

==== Synopsis

> ----

Reviewer: ```suggestion
[source, syntax]
----
```

Author: Basically none of the synopsis or example sections have this. I suggest to keep it somewhat consistent by not introducing it now and only here but later and everywhere. Wdyt?

Reviewer: Yeah fair enough!

Author: I created an issue (#43) to take care of that in a separate PR.

TELEMETRY api
----

.Example 1

> ----

Reviewer: ```suggestion
[source, Bolt]
----
```
And for all the others. This styles the codeblocks nicer (a bit stupid that the style doesn't get applied on "anonymous" blocks -- I guess it was a decision to avoid inadvertently styling stuff that shouldn't).

TELEMETRY 0
----

==== Server response `SUCCESS`

> If a `TELEMETRY` message request has been successfully received, the server should respond with a `SUCCESS` and remain in its current state.

Reviewer: ```suggestion
If a `TELEMETRY` message request is successfully received, the server responds with a `SUCCESS` and remains in its current state.
```
No `should` because it is a behavior that is out of the control of the client, and readers are not asked to implement behaviors on the server side. We can assume it _always_ happens what it _should_ happen.

Author: Fair point. I copy pasted this from another place. Wanna go over the whole docs and fix it as well? :)

Reviewer: Can't just now, but at some point I will :)

.Example
----
SUCCESS {}
----

==== Server response `FAILURE`

> If a `TELEMETRY` message contains a value that is not a valid `api` value or is sent in the wrong state, the server should respond with a `FAILURE` message and enter the `FAILED` state.

Reviewer: ```suggestion
If a `TELEMETRY` message contains a value that is not a valid `api` value or is sent in the wrong state, the server responds with a `FAILURE` message and enters the `FAILED` state.
```
